spec/bitcoin/wallet/utxo_handler_spec.rb

```diff
@@ -0,0 +1,162 @@
+require 'spec_helper'
+
+describe 'Bitcoin::Wallet::UtxoHandler' do
+  let(:handler) { Bitcoin::Wallet::UtxoHandler.new(spv, utxo_db) }
+  let(:utxo_db) { create_test_utxo_db }
+  let(:spv) { create_test_spv }
+  let(:wallet) { Bitcoin::Wallet::Base.create(1, 'tmp/wallet_db/') }
+
+  before { allow(spv).to receive(:wallet).and_return(wallet) }
+
+  after do
+    utxo_db.close
+    wallet.close
+    FileUtils.rm_r('tmp/wallet_db/')
+  end
+
+  context 'when node receives tx message' do
+    subject { handler.update(:tx, message) }
+
+    let(:funding_tx) do
+      # create transaction to send to the wallet
+      key = wallet.accounts.first.create_receive
+
+      # default script is native_segwit
+      script = Bitcoin::Script.to_p2wpkh(Bitcoin.hash160(key.pubkey))
```

Comment:
It may be prefer to derive the script from the derived `ExtPubkey`, as the process of generating the script from the account as above is quite possible. The developer should not specify which type of script (p2pkh, p2wpkh, p2sh-p2wpkh) is output, because the account already knows.